samples/file-encryptor/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
```

Comment:
So sorry... since you've found the real minimum required version, can you bump them all to 3.11? I'm fine if you want to bump that to another PR too.

Reply:
@anakrish IIRC we chose some version of cmake that's not the latest for compatibility with the cmake version that Visual Studio deploys on Windows. Is the bump to 3.11 consistent with that?

Reply:
3.12 is the version that we have locked down for OE due to Visual Studio.
https://github.com/Microsoft/openenclave/commit/8121f4606c73982566276cee89e4df256342319c.

Any reason why we want to choose a lesser cmake version for samples?

Reply:
My reasoning would be to make it as easy as possible to use the samples.

Reply:
Since Visual Studio comes with `3.12` (as per https://github.com/Microsoft/openenclave/commit/8121f4606c73982566276cee89e4df256342319c), I have bumped the version to `3.11` for the samples because that is the minimum we need for Linux (https://github.com/Microsoft/openenclave/issues/1561)

We can bump it to 3.12 if needed with a different PR.